tests/tests/regenesis.rs

```diff
@@ -0,0 +1,216 @@
+use std::path::Path;
+
+use clap::Parser;
+use fuel_core::{
+    chain_config::{
+        SnapshotMetadata,
+        SnapshotReader,
+    },
+    service::{
+        Config,
+        FuelService,
+        ServiceTrait,
+    },
+};
+use fuel_core_bin::cli::snapshot;
+use fuel_core_client::client::{
+    pagination::{
+        PageDirection,
+        PaginationRequest,
+    },
+    FuelClient,
+};
+use fuel_core_poa::Trigger;
+use fuel_core_types::{
+    fuel_tx::*,
+    fuel_vm::{
+        checked_transaction::IntoChecked,
+        *,
+    },
+};
+use rand::SeedableRng;
+use tempfile::{
+    tempdir,
+    TempDir,
+};
+
+pub struct FuelCoreDriver {
+    /// This must be before the db_dir as the drop order matters here
+    pub node: FuelService,
+    pub db_dir: TempDir,
+    pub client: FuelClient,
+}
+impl FuelCoreDriver {
+    pub async fn spawn(from_snapshot: Option<&Path>) -> anyhow::Result<Self> {
+        // Generate temp params
+        let db_dir = tempdir()?;
+
+        let mut config = Config::local_node();
+        config.debug = true;
+        config.block_production = Trigger::Instant;
+        config.combined_db_config.database_path = db_dir.path().to_path_buf();
```

Comment:
I still think it would be better to re-use the part of the `run::exec` command and use CLI arguments to init all what you need=) In this case, you will not duplicate the logic of `fuel-core-bin`.

Reply:
Hmmm I though you argued against doing that above. I think this approach here is quite clean. I'm not sure what logic is duplicarted here?

Reply:
<img width="912" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/8685e4b1-e42a-47b7-b79b-d0e30e18cfd4">

We can reuse the code from here. Instead of creating `Config::local_node()` we can get it from the command like `command.get_config()?`. As an example, we don't use `Config::local_node()` inside of the `fuel-core-bin` anymore, instead it uses config for the testnet=) So to avoid different behavior it is better to use part of the `run::exec`=)